src/openloco/things/CreateVehicle.cpp

```diff
@@ -0,0 +1,990 @@
+    {
+        switch (head->var_5D)
+        {
+            case 8:
+                gGameCommandErrorText = string_ids::vehicle_has_crashed;
+                return false;
+            case 9:
+                gGameCommandErrorText = string_ids::vehicle_is_stuck;
+                return false;
+            case 7:
+                gGameCommandErrorText = string_ids::vehicle_has_broken_down;
+                return false;
+            default:
+            {
+                auto veh1 = reinterpret_cast<openloco::vehicle*>(head)->next_car();
+                auto veh2 = veh1->next_car()->as_vehicle_2();
+                if (head->vehicleType == VehicleType::plane || head->vehicleType == VehicleType::ship)
+                {
+                    if (veh2->var_73 & (1 << 0))
+                    {
+                        gGameCommandErrorText = string_ids::vehicle_has_broken_down;
+                        return false;
+                    }
+
+                    if (head->tile_x == -1)
+                    {
+                        return true;
+                    }
+
+                    if (head->var_5D != 6 && head->var_5D != 1)
+                    {
+                        gGameCommandErrorText = string_ids::vehicle_must_be_stopped;
+                        return false;
+                    }
+                    if (veh2->var_56 == 0)
+                    {
+                        return true;
+                    }
+                    gGameCommandErrorText = string_ids::vehicle_must_be_stopped;
+                    return false;
+                }
+                else
+                {
+
+                    if (head->tile_x == -1)
+                    {
+                        return true;
+                    }
+                    if (veh2->var_56 == 0)
+                    {
+                        return true;
+                    }
+                    if (veh1->var_3C <= 13961)
+                    {
+                        return true;
+                    }
+                    gGameCommandErrorText = string_ids::vehicle_must_be_stopped;
+                    return false;
+                }
+            }
+        }
+    }
+
+    static void sub_4B08DD(openloco::vehicle_head* const head)
+    {
+        registers regs{};
+        regs.esi = reinterpret_cast<uint32_t>(head);
+        call(0x004B08DD, regs);
+    }
+
+    // 0x00470039
+    static openloco::vehicle_base* createVehicleThing()
+    {
+        registers regs{};
+        call(0x00470039, regs);
+        return reinterpret_cast<openloco::vehicle_base*>(regs.esi);
+    }
+
+    template<typename T>
+    static T* createVehicleThing()
+    {
+        auto* const base = createVehicleThing();
+        base->base_type = thing_base_type::vehicle;
+        base->type = T::VehicleThingType;
+        return reinterpret_cast<T*>(base);
+    }
+
+    static void sub_4BA873(openloco::vehicle_bogie* const vehBogie)
```

Comment:
Too math intensive for me. Someone else validate please.

Reply:
L214 looks incorrect should be `/ 65536`